llama-index-core/llama_index/core/settings.py

```diff
@@ -197,43 +200,6 @@ def text_splitter(self, text_splitter: NodeParser) -> None:
         """Set the text splitter."""
         self.node_parser = text_splitter
 
-    # ---- Prompt helper ----
-
-    @property
-    def prompt_helper(self) -> PromptHelper:
-        """Get the prompt helper."""
-        if self._llm is not None and self._prompt_helper is None:
-            self._prompt_helper = PromptHelper.from_llm_metadata(self._llm.metadata)
-        elif self._prompt_helper is None:
-            self._prompt_helper = PromptHelper()
-
-        return self._prompt_helper
-
-    @prompt_helper.setter
-    def prompt_helper(self, prompt_helper: PromptHelper) -> None:
-        """Set the prompt helper."""
-        self._prompt_helper = prompt_helper
-
-    @property
-    def num_output(self) -> int:
-        """Get the number of outputs."""
-        return self.prompt_helper.num_output
-
-    @num_output.setter
-    def num_output(self, num_output: int) -> None:
-        """Set the number of outputs."""
-        self.prompt_helper.num_output = num_output
-
-    @property
-    def context_window(self) -> int:
```

Comment:
wait why is this removed

Reply:
It doesn't need to be a getter/setter now, since the prompt helper is initialized every time from metadata

Reply:
I added them as optional attributes on the settings itself

Reply:
Maybe this is a tad too radical of a change. Debating walking this back a tad lol

Reply:
oh i missed that. i think the only catch is it makes typing kind of annoying because num_output is now an Optional type

Reply:
Walked this back slightly -- but still in a way that fixes the initial bug :) Someday we will rip out prompt helper